### PR TITLE
Run stage 0 std tests in CI

### LINKED ISSUE #95996: Run `test --stage 0 library/std` in CI
These tests were recently broken by https://github.com/rust-lang/rust/pull/95449. It shouldn't be possible for these to be broken on master - nearly everyone who contributes to the standard library uses `--stage 0`, and this makes it very hard to verify changes.

We should get these running in mingw-check or some other builder.

@rustbot label +P-high +A-testsuite +T-libs

_Originally posted by @Mark-Simulacrum in https://github.com/rust-lang/rust/issues/95993#issuecomment-1097491476_

### PR DESCRIPTION
**Pull request by ghost**
Fixes https://github.com/rust-lang/rust/issues/95996

### CODE REVIEW

**Comment by gimbles** [comment]
Fixes #95996

File: src/ci/docker/host-x86_64/mingw-check/Dockerfile [hunk 1]
```diff
@@ -36,6 +36,7 @@ ENV SCRIPT python3 ../x.py --stage 2 test src/tools/expand-yaml-anchors && \
            python3 ../x.py build --stage 0 src/tools/build-manifest && \
            python3 ../x.py test --stage 0 src/tools/compiletest && \
            python3 ../x.py test --stage 2 src/tools/tidy && \
+           python3 ../x.py test --stage 0 library/std && \
```
Comment (jyn514) [inline src/ci/docker/host-x86_64/mingw-check/Dockerfile:39]: This just tests the std crate and not any others. Can you also test the other crates?
```suggestion
           python3 ../x.py test --stage 0 core alloc std test proc_macro && \
```

**Comment by pietroalbini** [comment]
@bors r+

Thanks!

**Comment by rylev** [comment]
The regression is just [noise correction from the previous run](https://perf.rust-lang.org/compare.html?start=db41351753df840773ca628d8daa040e95d00eef&end=1cd72b734318720adb99dc72147bb8169ef76ffe&stat=instructions:u).

@rustbot label: +perf-regression-triaged

**Pull request merged**